tests/e2e/test_s2ef.py

```diff
@@ -0,0 +1,409 @@
+import collections.abc
+import copy
+import glob
+import os
+import tempfile
+from pathlib import Path, PosixPath
+from typing import List
+
+import numpy as np
+import pytest
+import torch
+import yaml
+from tensorboard.backend.event_processing.event_accumulator import (
+    EventAccumulator,
+)
+
+from main import Runner
+from ocpmodels.common.utils import build_config, setup_logging
+from ocpmodels.datasets import TrajectoryLmdbDataset
+from ocpmodels.models.escn.so3 import CoefficientMapping
+from ocpmodels.models.escn.so3 import SO3_Embedding as escn_SO3_Embedding
+from ocpmodels.trainers import OCPTrainer
+
+from ocpmodels import models  # isort: skip
+from ocpmodels.common import logger  # isort: skip
+
+
+setup_logging()
+
+
+class DotDict(dict):
+    __getattr__ = dict.__getitem__
+
+
+@pytest.fixture
+def base_command_line_args():
+    return DotDict(
+        {
+            "mode": "train",
+            "identifier": "",
+            "debug": False,
+            "run_dir": "./",
+            "print_every": 10,
+            "seed": 100,
+            "amp": False,
+            "checkpoint": None,
+            "timestamp_id": None,
+            "sweep_yml": None,
+            "distributed": False,
+            "submit": False,
+            "summit": False,
+            "logdir": PosixPath("logs"),
+            "slurm_partition": "ocp",
+            "slurm_mem": 80,
+            "slurm_timeout": 72,
+            "local_rank": 0,
+            "distributed_port": 0,
+            "world_size": 1,
+            "noddp": False,
+            "gp_gpus": 0,
+            "cpu": True,
+            "num_nodes": 0,
+            "num_gpus": 0,
+            "distributed_backend": "nccl",
+            "no_ddp": False,
+        }
+    )
+
+
+@pytest.fixture
+def configs():
+    return {
+        "escn": Path("tests/models/test_configs/test_escn.yml"),
+        "gemnet": Path("tests/models/test_configs/test_gemnet.yml"),
+        "equiformer_v2": Path(
+            "tests/models/test_configs/test_equiformerv2.yml"
+        ),
+    }
+
+
+@pytest.fixture
+def tutorial_train_src(tutorial_dataset_path):
+    return tutorial_dataset_path / "s2ef/train_100"
+
+
+@pytest.fixture
+def tutorial_val_src(tutorial_dataset_path):
+    return tutorial_dataset_path / "s2ef/val_20"
+
+
+def oc20_lmdb_train_config_from_path(src):
```

Comment:
looks like this is only used once in oc20_lmdb_train_and_val_from_paths below, itll be easier to read the code if this dictionary is explicitly defined in there rather than this helper function

Reply:
definitely fixed!, some of these functions were just not properly cleaned up after a long battle with pytest. I think they look better now